com.community.netcode.extensions/Runtime/NetworkDictionary/NetworkDictionary.cs

```diff
@@ -474,9 +474,15 @@ public bool TryGetValue(TKey key, out TValue value)
         private void HandleAddDictionaryEvent(NetworkDictionaryEvent<TKey, TValue> dictionaryEvent)
         {
             m_DirtyEvents.Add(dictionaryEvent);
+            MarkNetworkObjectDirty();
             OnDictionaryChanged?.Invoke(dictionaryEvent);
         }
 
+        internal void MarkNetworkObjectDirty()
```

Comment:
Yeah, that seems like something that would be required after we added the DirtyObjects optimization. It almost as if we had completely forgotten to go and patch the contribution part. 

Thanks for pointing this out.